framework/include/problems/FEProblemBase.h

```diff
@@ -611,6 +612,13 @@ class FEProblemBase : public SubProblem, public Restartable
   addDistribution(const std::string & type, const std::string & name, InputParameters & parameters);
   virtual Distribution & getDistribution(const std::string & name);
 
+  /**
+   * The following functions will enable MOOSE to have the capability to import likelihoods
+   */
+  virtual void
+  addLikelihood(const std::string & type, const std::string & name, InputParameters & parameters);
+  virtual Likelihood & getLikelihood(const std::string & name);
+
```

Comment:
I am against this. Likelihood and its handling should go to the STM. You can make the likelihood a `MooseObject` and handle it just as the `Covariance` object. If you need functions like execute, make it a `UserObject`.